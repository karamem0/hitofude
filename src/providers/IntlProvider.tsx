//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import { IntlProvider as Provider, createIntl } from 'react-intl';
import en from '../translations/en.json';
import ja from '../translations/ja.json';

const translations: { [key: string]: Record<string, string> } = {
  en,
  ja
};

const intl = createIntl({
  defaultLocale: 'en',
  locale: window.navigator.language.substring(0, 2),
  messages: translations[window.navigator.language]
});

function IntlProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  return (
    <Provider {...intl}>
      {children}
    </Provider>
  );

}

export default IntlProvider;
