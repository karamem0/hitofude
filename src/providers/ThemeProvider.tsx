//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import {
  FluentProvider as Provider,
  Theme
} from '@fluentui/react-components';

import { darkTheme, lightTheme } from '../themes/Theme';
import { InvalidOperationError } from '../types/Error';
import { ThemeName } from '../types/Model';

interface ThemeState {
  theme: Theme,
  themeName: ThemeName
}

interface ThemeContextState {
  theme: ThemeState,
  changeTheme: (themeName: ThemeName) => void
}

const ThemeContext = React.createContext<ThemeContextState | undefined>(undefined);

export const useTheme = (): ThemeContextState => {
  const value = React.useContext(ThemeContext);
  if (value == null) {
    throw new InvalidOperationError();
  }
  return value;
};

function ThemeProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  const [ theme, setTheme ] = React.useState<ThemeState>({
    theme: lightTheme,
    themeName: ThemeName.light
  });

  const handleChangeTheme = React.useMemo(() => (value: ThemeName) => {
    switch (value) {
      case ThemeName.light:
        setTheme(() => ({
          theme: lightTheme,
          themeName: value
        }));
        break;
      case ThemeName.dark:
        setTheme(() => ({
          theme: darkTheme,
          themeName: value
        }));
        break;
      default:
        break;
    }
  }, []);

  const value = React.useMemo<ThemeContextState>(() => ({
    theme,
    changeTheme: handleChangeTheme
  }), [
    theme,
    handleChangeTheme
  ]);

  return (
    <ThemeContext.Provider value={value}>
      <Provider theme={value.theme.theme}>
        {children}
      </Provider>
    </ThemeContext.Provider>
  );

}

export default ThemeProvider;
