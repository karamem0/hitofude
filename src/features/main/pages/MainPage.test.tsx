//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import IntlProvider from '../../../providers/IntlProvider';
import Presenter from './MainPage.presenter';
import ThemeProvider from '../../../providers/ThemeProvider';
import { render } from '@testing-library/react';

vi.mock('../components/AppBar', () => ({
  default: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="test-AppBar">
      {children}
    </div>
  )
}));

vi.mock('../components/AppTab', () => ({
  default: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="test-AppTab">
      {children}
    </div>
  )
}));

vi.mock('../components/AppContent', () => ({
  default: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="test-AppContent">
      {children}
    </div>
  )
}));

vi.mock('../components/MeControl', () => ({
  default: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="test-MeControl">
      {children}
    </div>
  )
}));

vi.mock('../../dialog/factories/DialogFactory', () => ({
  default: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="test-DialogFactory">
      {children}
    </div>
  )
}));

vi.mock('../../panel/factories/SidePanelFactory', () => ({
  default: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="test-SidePanelFactory">
      {children}
    </div>
  )
}));

vi.mock('../../../common/components/ErrorNotification', () => ({
  default: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="test-ErrorNotification">
      {children}
    </div>
  )
}));

it('should match the snapshot', () => {
  const params = {};
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
