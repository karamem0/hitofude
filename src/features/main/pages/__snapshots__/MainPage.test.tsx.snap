// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: column;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
}

.emotion-1 {
  position: fixed;
  z-index: 200;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
  height: 3rem;
  padding: 0 1rem;
  background-color: #795c5c;
}

.emotion-2 {
  font-size: 24px;
  color: #ffffff;
  text-align: center;
}

.emotion-3 {
  display: grid;
  height: calc(100svh - 2.5rem);
  margin: 2.5rem 0 0;
}

@media all and (width <= 960px) {
  .emotion-3 {
    grid-template-rows: auto;
    grid-template-columns: auto 1fr;
  }

  .emotion-3>div:last-of-type {
    position: fixed;
    width: calc(100vw - 3.5rem);
    height: calc(100svh - 2.5rem);
    margin: 0 0 0 3.5rem;
  }
}

@media not all and (width <= 960px) {
  .emotion-3 {
    grid-template-rows: auto;
    grid-template-columns: auto auto 1fr;
  }

  .emotion-3>div:last-of-type {
    position: static;
    margin: 0;
  }
}

<div
    class="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      data-testid="test-ErrorNotification"
    />
    <div
      data-testid="test-DialogFactory"
    />
    <div
      data-testid="test-SidePanelFactory"
    />
    <div
      class="emotion-0"
    >
      <header
        class="emotion-1"
      >
        <h1
          class="fui-Text emotion-2 ___c56hd20_1xy0w11 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
        >
          Hitofude
        </h1>
        <div
          data-testid="test-MeControl"
        />
      </header>
      <main
        class="emotion-3"
      >
        <div
          data-testid="test-AppBar"
        />
        <div
          data-testid="test-AppTab"
        />
        <div
          data-testid="test-AppContent"
        />
      </main>
    </div>
  </div>
</DocumentFragment>
`;
